tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/central.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/conn.h"
+
+#include <stdint.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+void adv_new_ident(void)
+{
+	/* create a new identity:
+	 * the stack disallows connecting to the same peer twice
+	 */
+	int id_a = bt_id_create(NULL, NULL);
+
+	ASSERT(id_a >= 0, "bt_id_create id_a failed (err %d)\n", id_a);
+	advertise_connectable(id_a, 0);
+}
+
+void central_procedure(void)
+{
+	bs_bt_utils_setup();
+	k_msleep(1000);
+
+	scan_connect_to_first_result();
+	wait_connected();
+	/* DUT is peripheral */
+
+	adv_new_ident();
+	wait_connected();
+	/* DUT is central & peripheral */
+
+	k_sleep(K_SECONDS(5));
+
+	printk("disconnect peripheral\n");
+	disconnect(BT_CONN_ROLE_PERIPHERAL);
+	/* DUT starts scanning again (using NRPA) */
+
+	k_sleep(K_SECONDS(5));
+
+	printk("disconnect central\n");
+	disconnect(BT_CONN_ROLE_CENTRAL);
+
+	/* DUT starts advertising again (with scanner's NRPA)
+	 * should give wrong address
+	 */
+	scan_expect_same_address();
+	k_sleep(K_SECONDS(1));
```

Comment:
Why is this sleep needed?

Reply:
To leave some time for the scanner to get at least one report.

Reply:
IMO it would be better to use a flag for this, and wait for the flag to be set. Arbitrary timer values such as this makes it much more difficult to follow the code, and to know why we are waiting.

Reply:
removed